core/src/main/java/se/tre/freki/query/predicate/TimeSeriesQueryPredicate.java

```diff
@@ -56,6 +56,12 @@ public void addTagPredicate(final TimeSeriesTagPredicate tagPredicate) {
     public TimeSeriesQueryPredicate build() {
       checkState(metric != null);
 
+      try {
+        checkState(tagPredicates.size() > 0);
```

Comment:
Use a regular if condition and throw the `QueryException` directly instead of catching the potential exception from `checkState`. `checkState` doing it this way will hide where the exception is thrown since you don't send the exception parameter `e` along as the cause to the `QueryException` and throwing and catching an exception is generally considered bad practice.